@xen-orchestra/proxy/src/app/_debounceResource.js

```diff
@@ -12,24 +14,23 @@ export async function debounceResource(pDisposable, hooks, delay = 0) {
     if (timeoutId !== undefined) {
       clearTimeout(timeoutId)
       timeoutId = undefined
-      hooks.removeListener('stop', onStop)
 
       return disposable.dispose()
     }
   }
 
-  const onStop = () => {
+  disposers.push(() => {
     const shouldDisposeNow = timeoutId !== undefined
     if (shouldDisposeNow) {
       return disposeWrapper()
     } else {
       // will dispose ASAP
       delay = 0
     }
-  }
-  hooks.on('stop', onStop)
+  })
 
   return new Disposable(disposable.value, () => {
     timeoutId = setTimeout(disposeWrapper, delay)
   })
 }
+debounceResource.flushAll = () => asyncMap(disposers, dispose => dispose())
```

Comment:
Why `asyncMap`?

Reply:
To wait for all disposers to finish..

Reply:
My bad, I didn't see your disposers were really async.